packages/core/src/typeschema/validation.ts

```diff
@@ -316,27 +314,6 @@ function isChoiceOfType(
   return false;
 }
 
-function getNestedProperty(value: TypedValue, key: string): (TypedValue | TypedValue[] | undefined)[] {
-  const [firstProp, ...nestedProps] = key.split('.');
```

Comment:
I ran a sanity test on this  `if (key.includes('.')) throw new Error('period');` and it didn't throw

Up above, it looks like you're already doing the key splitting and path management.  This function never enters into the `for` loop.